### PR TITLE
Javascript Capstone Project Movie

### PR DESCRIPTION
**Pull request by leloufadel**
This pull request includes the implementation of the Home Page and Comments Popup for the web application. The following tasks have been completed:

**General Requirements:**

-Linter errors have been resolved to ensure code quality.
-Gitflow has been followed correctly throughout the development process.
-HTML/CSS Best Practices has been followed

**Project Requirements:**

The chosen API, for TV-MAZE, has been integrated into the web application.
The Involvement API has been utilized to record user interactions such as likes and comments.
**Interfaces:**

The following interfaces have been built:
Home Page: Loads data from the selected API, displaying a list of items and item likes. Interaction recording for likes is implemented.
Comments Popup: Loads data from the selected API, displaying details about the selected item and item comments. Comment submission functionality has been implemented.

**Counters:**

Counters have been implemented in all interfaces to display the following:
Number of items on the home page.
Number of comments on the comments popup.
Specific functions have been created to calculate these numbers, and unit tests using Jest have been written to ensure their accuracy.

**Technical Set Up:**

The repository has been set up on GitHub, following the Gitflow workflow.
Webpack has been configured for the project.
Jest, a JavaScript testing library, has been set up to facilitate unit testing.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+</head>
+<body>
+  <header>
+    <h1>MovieShow</h1>
+    <nav>
+      <div class="desktop-menu">
+        <ul>
+          <li id="all">All Movie <span class="counter movies"></span></li>
+          <li>planets</li>
+          <li>Races</li>
```
Comment (fobadara) [inline src/index.html:20]: [OPTIONAL] I don't think you should include this since there are not doing anything.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    body: raw,
+    redirect: 'follow',
+  };
+
+  fetch(
+    `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${APP_ID}/likes`,
+    requestOptions
+  )
+    .then((response) => response.text())
+    .then((result) => console.log(result))
```
Comment (fobadara) [inline src/index.js:55]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    .catch((error) => console.log('error', error));
```
Comment (fobadara) [inline src/index.js:56]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    const likesCounterElement = document.querySelector('.counter.likes');
+    const moviesCounterElement = document.querySelector('.counter.movies');
+    const allMoviesCounterElement = document.querySelector('#all .counter');
+
+    commentsCounterElement.textContent = commentsCount;
+    likesCounterElement.textContent = likesCount;
+    moviesCounterElement.textContent = moviesCount;
+    allMoviesCounterElement.textContent = moviesCount;
+  })
+  .catch((err) => console.log(err));
```
Comment (fobadara) [inline src/index.js:118]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

File: src/modules/modal.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+   name="comment"
+   rows="4" cols="40"
+   placeholder="Enter your comment here..."></textarea><br>
+   <input type="submit" value="comment" class="comments">
+</form>
+
+  `;
+  // Get the <span> element that closes the modal
+  const span = document.querySelector('.close');
+  console.log(span);
```
Comment (fobadara) [inline src/modules/modal.js:62]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

File: src/modules/modal.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  // When the user clicks the button, open the modal
+  modal.style.display = 'block';
+
+  // When the user clicks on <span> (x), close the modal
+  span.onclick = () => {
+    // console.log(span);
```
Comment (fobadara) [inline src/modules/modal.js:69]: [REQUIRED] Kindly remove the commented out code to make your project more professional

File: src/modules/post-comments.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+const APP_ID = 'DUanzoHMk8l8HLimHh6p';
+
+export const postComments = async (username, comment, itemId) => {
+  const myHeaders = new Headers();
+  myHeaders.append('Content-Type', 'application/json');
+  myHeaders.append('Authorization', 'Basic bWFqZDowMDAw');
+  console.log(itemId);
```
Comment (fobadara) [inline src/modules/post-comments.js:7]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

File: src/modules/post-comments.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    body: raw,
+    redirect: 'follow',
+  };
+  fetch(
+    `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${APP_ID}/comments`,
+    requestOptions
+  )
+    .then((response) => response.text())
+    .then((result) => console.log(result))
+    .catch((error) => console.log('error', error));
```
Comment (fobadara) [inline src/modules/post-comments.js:25]: [REQUIRED] Kindly remove all `console.log` from your code in order to make your project more professional

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Clean code
✅ Uses try-and-catch blocks for data fetching
✅ Descriptive README

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by majdalkilany** [comment]
remove all consoles from the code and remove the li elements from the index.html page

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👏 You did well to remove `console.log` statements and commented codes, the project codebase now looks clean.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**